### ISSUE TITLE
cloneDeep doesn't copy read-only values

### ISSUE THREAD

**Issue by Xiot** [toss/es-toolkit#702]
`cloneDeep`, both the 'normal' and 'compat' version, don't copy read-only properties.

```typescript
const obj = {}
    Object.defineProperties(obj, {
      first: {
        enumerable: true,
        writable: true,
        value: 4,
      },
      second: {
        enumerable: true,
        get() {
          return 3
        },
      },
    })    
    obj.third = 9

  console.log('original', obj)
  console.log('cloneDeep', cloneDeep(obj)
  console.log('assign', Object.assign({}, obj) 
```

output
```
original { first: 4, second: [Getter], third: 9 }
cloneDeep { first: 4, third: 9 }
assign { first: 4, second: 3, third: 9 }
```

The actual issue is in `copyProperties`
https://github.com/toss/es-toolkit/blob/main/src/object/cloneDeep.ts#L203-L207

specifically
```typescript
    const descriptor = Object.getOwnPropertyDescriptor(source, key);

    if (descriptor?.writable || descriptor?.set) {
      target[key] = cloneDeepImpl(source[key], stack);
    }
```

Here is it looping over the properties on `source`, but it only copies values from the source whose properties are `writable` or have a `set` function.

This is probably backwards, since it is actually reading from the `source` and writing to the `target`.

That conditional should actually be
```typescript
if (descriptor?.enumerable) { ... }
```

Actually, you don't even need that conditional at all since `Object.keys` won't return keys for non-enumerable properties.

**Comment by raon0211** [comment]
Thanks for your thorough review of the logic. We'll make sure to address it in the next release.

**Comment by raon0211** [comment]
This was resolved in es-toolkit v1.25.0. Please refer to [this commit](https://github.com/toss/es-toolkit/commit/5b60dbc7d26132a046e17b4ffc02931fa30b9e8b) for details. Thanks for finding the issue in our library!
